Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -1535,6 +1535,13 @@ static bool isCopyDisplayedPixelBufferAvailable()
     m_decompressionSession = WebCoreDecompressionSession::createOpenGL();
     m_decompressionSession->setTimebase([m_synchronizer timebase]);
     m_decompressionSession->setResourceOwner(m_resourceOwner);
+    m_decompressionSession->setErrorListener([weakThis = WeakPtr { *this }](OSStatus) {
+        if (RefPtr protectedThis = weakThis.get()) {
+            protectedThis->setNetworkState(MediaPlayer::NetworkState::DecodeError);
+            protectedThis->setReadyState(MediaPlayer::ReadyState::HaveNothing);
```

Comment:
Why not setReadyState in MediaPlayerPrivateMediaSourceAVFObjC.mm?
Also, can you beef up the commit log, it is not clear to me why the error listener is only needed for media elements that are not in the DOM.

Reply:
Because when it's not in the DOM we use a WebCoreDecompressionSession if we have an AVSBDL, the error are handled by the SourceBufferPrivateAVFObjC listener to the NotificationCentre for errors.

As for why the readyState isn't changed for MSE, I made it consistent on how the handled are managed with the MSE player.

For the WebM player, it's probably not required, but I didn't want to introduce a different behaviour, so it does the same as how other decoding errors are handled.